pkg/manifest/locator.go

```diff
@@ -0,0 +1,190 @@
+package manifest
+
+import (
+	"encoding/json"
+
+	"github.com/pkg/errors"
+)
+
+// One or more alternative expressions of the location.
+// https://github.com/readium/architecture/tree/master/models/locators#the-location-object
+type Locations struct {
+	Fragments        []string               `json:"fragments,omitempty"`        // Contains one or more fragment in the resource referenced by the [Locator].
+	Progression      float64                `json:"progression,omitempty"`      // Progression in the resource expressed as a percentage (between 0 and 1).
+	Position         uint                   `json:"position,omitempty"`         // An index in the publication (>= 1).
+	TotalProgression float64                `json:"totalProgression,omitempty"` // Progression in the publication expressed as a percentage (between 0 and 1).
+	OtherLocations   map[string]interface{} `json:"otherLocations"`             // Additional locations for extensions.
+}
+
+func LocationsFromJSON(rawJson map[string]interface{}) (*Locations, error) {
```

Comment:
Using value semantics could be more convenient and safer for the `Locator` models.
```suggestion
func LocationsFromJSON(rawJson map[string]interface{}) (Locations, error) {
```

Reply:
This is to deal with the case where the raw JSON being inputted is nil. Would you prefer nil JSON become a default-initialized Locations object? That *may* result in some empty objects where they shouldn't be (or if unmarshalling and them marshalling again). Note this only applies because it's part of the unmarshalling/marshalling logic, normally I would agree values are safer

Reply:
I think it depends on what we decide with https://github.com/readium/go-toolkit/pull/61#discussion_r918430208